challenge/contact.py

```diff
@@ -1,188 +0,0 @@
-#!/usr/bin/env python
-#
-from selenium import webdriver
-from urllib.parse import urlparse,urlunparse
-import urllib.request
-
-import logging
-import pdb
-import re
-import sys
-import platform
-
-class ui:
-    """ web access with ui capabilities """
-    def __init__(self,hack=''):
-        """ Selenium web driver can be set up 'headless' browser on some remote system, which could improve performance or allow running multiple copies"""
-        if hack:
-            self.web = webdriver.Firefox(webdriver.FirefoxProfile(hack))
-        else:
-            self.web = webdriver.Firefox()
-
-        self.actions = webdriver.common.action_chains.ActionChains(self.web)
-        #self.scan.actions.getCurrentWindow().focus()
-        for _ in range(12):
```

Comment:
There was s comment related to this in the 'development.txt' file.  Since there is some UI interaction required to see anything on the top 'jana.com' page (content loading via trigger on scrolling), I really wanted to just scroll down some distance, but PageDown required setting focus somewhere first, since the browser address bar initially has focus.  Just doing a bunch of tabs seemed easier at the time than reading a lot of selenium documentation to figure out how to do this right.  (I have no prior experience with selenium)

12 was arrived at empirically, by watching focus cycle on various pages.  There is always a chance that some diabolical JavaScript programmer can hide the data you want, so at some point there is a trade-off with the amount of effort put into the scraping tool and the performance hit of jumping around and triggering lots of new ad/tracking requests.